pkg/agent/route/route_linux.go

```diff
@@ -822,21 +1063,13 @@ func (c *Client) restoreIptablesData(podCIDR *net.IPNet,
 		writeLine(iptablesData, iptables.MakeChainLine(antreaPreRoutingChain))
 	}
 	if c.proxyAll {
-		writeLine(iptablesData, []string{
-			"-A", antreaPreRoutingChain,
-			"-m", "comment", "--comment", `"Antrea: DNAT external to NodePort packets"`,
-			"-m", "set", "--match-set", nodePortIPSet, "dst,dst",
-			"-j", iptables.DNATTarget,
-			"--to-destination", nodePortDNATVirtualIP.String(),
-		}...)
+		for _, ruleStr := range proxyAllNatIPTables[antreaPreRoutingChain] {
+			writeLine(iptablesData, ruleStr)
+		}
 		writeLine(iptablesData, iptables.MakeChainLine(antreaOutputChain))
-		writeLine(iptablesData, []string{
-			"-A", antreaOutputChain,
-			"-m", "comment", "--comment", `"Antrea: DNAT local to NodePort packets"`,
-			"-m", "set", "--match-set", nodePortIPSet, "dst,dst",
-			"-j", iptables.DNATTarget,
-			"--to-destination", nodePortDNATVirtualIP.String(),
-		}...)
+		for _, ruleStr := range proxyAllNatIPTables[antreaOutputChain] {
+			writeLine(iptablesData, ruleStr)
+		}
```

Comment:
I think introducing the variable `proxyAllNatIPTables` here makes the code less consolidated and consistent.
Previously we see the rules in the function itself, now some rules are constructed in another function in another form, and it uses a map to store the results and consume the map in this function dispersedly. 

Furthermore, it's very confusing to see the comments and code around L1126, where the comment is not really explaining the code which is just about a vague map:
```
// If AntreaProxy full support is enabled, it SNATs the packets whose source IP is VirtualServiceIPv4/VirtualServiceIPv6
// so the packets can be routed back to this Node.
if c.proxyAll {
	for _, ruleStr := range proxyAllNatIPTables[antreaPostRoutingChain] {
		writeLine(iptablesData, ruleStr)
	}
}
```

I would avoid such partial refactoring, and I don't see advantage to extract a function to construct rules for proxyAll mode separately.

Reply:
Will refactor this part to be consistent with the existing code.